### ISSUE TITLE
iOS implementation of Authorization considers 201 (Created) http code as a failure

### ISSUE THREAD

**Issue by cadicetienne** [transistorsoft/flutter_background_geolocation#673]
## Your Environment
* Plugin version: 4.3.0
* Platform: iOS 
* OS version: 13.1.2
* Device manufacturer / model: iPhone 7
* Flutter info (`flutter doctor`):

``` 
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6.1 20G224 darwin-x64, locale fr-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.62.2)
[✓] Connected device (3 available)

• No issues found!
``` 

* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
╔═════════════════════════════════════════════
║ TSLocationManager (build 380)
╠══════════════════════════════════════════════
{
    activityRecognitionInterval = 10000;
    activityType = 1;
    authorization =     {
        accessToken = "eyJhb<redacted>";
        expires = "-1";
        refreshPayload =         {
            "refresh_token" = "{refreshToken}";
        };
        refreshToken = "2f3e2<redacted>";
        refreshUrl = "https://staging.api.h2s.alyce.fr/api/v1/auth/refresh";
        strategy = jwt;
    };
    autoSync = 1;
    autoSyncThreshold = 10;
    batchSync = 1;
    debug = 0;
    desiredAccuracy = "-1";
    desiredOdometerAccuracy = 100;
    didDeviceReboot = 0;
    didLaunchInBackground = 0;
    didRequestUpgradeLocationAuthorization = 1;
    disableAutoSyncOnCellular = 0;
    disableElasticity = 0;
    disableLocationAuthorizationAlert = 0;
    disableMotionActivityUpdates = 0;
    disableStopDetection = 0;
    distanceFilter = 20;
    elasticityMultiplier = 1;
    enableTimestampMeta = 0;
    enabled = 0;
    extras =     {
    };
    geofenceInitialTriggerEntry = 1;
    geofenceProximityRadius = 2000;
    geofenceTemplate = "{\n          \"latitude\": <%=latitude%>,\n          \"longitude\": <%=longitude%>,\n          \"speed\": <%=speed%>,\n          \"speed_accuracy\": <%=speed_accuracy%>,\n          \"heading\": <%=heading%>,\n          \"heading_accuracy\": <%=heading_accuracy%>,\n          \"accuracy\": <%=accuracy%>,\n          \"altitude_accuracy\": <%=altitude_accuracy%>,\n          \"altitude\": <%=altitude%>,\n          \"altitude_accuracy\": <%=altitude_accuracy%>,\n          \"timestamp\": \"<%=timestamp%>\",\n          \"event\": \"<%=event%>\",\n          \"odometer\": <%=odometer%>,\n          \"activity_type\": \"<%=activity.type%>\",\n          \"activity_confidence\": <%=activity.confidence%>,\n          \"battery_level\": <%=battery.level%>,\n          \"battery_is_charging\": <%=battery.is_charging%>,\n          \"is_moving\": <%=is_moving%>\n          }";
    headers =     {
    };
    heartbeatInterval = 60;
    httpRootProperty = ".";
    httpTimeout = 60000;
    iOSHasWarnedLocationServicesOff = 0;
    isFirstBoot = 0;
    isMoving = 0;
    lastLocationAuthorizationStatus = 3;
    locationAuthorizationAlert =     {
        cancelButton = Annuler;
        instructions = "Vous devez choisir 'Toujours' dans le service de geolocalisation";
        settingsButton = "Param\U00e8tres";
        titleWhenNotEnabled = "Service de geolocalisation non activ\U00e9";
        titleWhenOff = "Service de geolocalisation ETEINT";
    };
    locationAuthorizationRequest = Always;
    locationTemplate = "{\n          \"latitude\": <%=latitude%>,\n          \"longitude\": <%=longitude%>,\n          \"speed\": <%=speed%>,\n          \"speed_accuracy\": <%=speed_accuracy%>,\n          \"heading\": <%=heading%>,\n          \"heading_accuracy\": <%=heading_accuracy%>,\n          \"accuracy\": <%=accuracy%>,\n          \"altitude_accuracy\": <%=altitude_accuracy%>,\n          \"altitude\": <%=altitude%>,\n          \"altitude_accuracy\": <%=altitude_accuracy%>,\n          \"timestamp\": \"<%=timestamp%>\",\n          \"event\": \"<%=event%>\",\n          \"odometer\": <%=odometer%>,\n          \"activity_type\": \"<%=activity.type%>\",\n          \"activity_confidence\": <%=activity.confidence%>,\n          \"battery_level\": <%=battery.level%>,\n          \"battery_is_charging\": <%=battery.is_charging%>,\n          \"is_moving\": <%=is_moving%>\n          }";
    locationTimeout = 60;
    locationsOrderDirection = ASC;
    logLevel = 5;
    logMaxDays = 3;
    maxBatchSize = 50;
    maxDaysToPersist = 1;
    maxRecordsToPersist = "-1";
    method = POST;
    minimumActivityRecognitionConfidence = 70;
    odometer = "37.6767424155458";
    params =     {
    };
    pausesLocationUpdatesAutomatically = 1;
    persistMode = 2;
    preventSuspend = 0;
    schedule =     (
    );
    schedulerEnabled = 0;
    showsBackgroundLocationIndicator = 0;
    startOnBoot = 1;
    stationaryRadius = 25;
    stopAfterElapsedMinutes = "-1";
    stopDetectionDelay = 0;
    stopOnStationary = 0;
    stopOnTerminate = 0;
    stopTimeout = 5;
    trackingMode = 1;
    url = "https://staging.api.h2s.alyce.fr/api/location";
    useSignificantChangesOnly = 0;
}
```

## Expected Behavior
When trying to refresh the token, a 201 response from the server should be considered as accepted (we create a new token, so some frameworks automatically return a 201).

## Actual Behavior
Instead, the iOS impl considers the status code != 200 as an error 

## Steps to Reproduce

1. Configure a server app or a mock server to have an access_token lifetime of a few secs/minutes
2. Configure a server app or a mock server returning 201 when refreshing the tokens
3. Configure the sdk Authorization.refreshUrl to target this app/mock server
4. Log in your app (and get the access & refresh token)
5. Wait for 1 minute (or the expiracy time you set server side)
6. Start the sdk (should send one event & trigger 401 error from the server)
7. Observe in the logs the failed attempt to refresh the token

## Context

Trying to have the authentication mechanism working on iOS.

## Debug logs
<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
-[TSAuthorization resolve:success:failure:]_block_invoke :clé: Token refresh failed to receive token from host staging.api.h2s.alyce.fr. HTTP STATUS: 201
```

</details>

**Comment by christocracy** [comment]
That's easy enough to fix.

```ojb-c
switch(statusCode) {
            case 200: {
                NSDictionary* data = [NSJSONSerialization JSONObjectWithData:responseData options:0 error:&error];
                if (error != nil) {
                    DDLogError(@"🔑 Failed to parse refresh-token response: %@", error.localizedDescription);
                    failure([[TSAuthorizationEvent alloc] initWithError:error]);
                    return;
                }
                TSAuthorizationEvent *event = [self refreshAuthorization:data];
                if (event.error != nil) {
                    DDLogWarn(@"🔑 Token refresh failed from host %@", [url host]);
                    failure(event);
                } else {
                    DDLogDebug(@"🔑 Refreshed access-token from host %@, %@", [url host], data);
                    success(event);
                }
                break;
            }
            default: {
                NSString *msg = [NSString stringWithFormat:@"Token refresh failed to receive token from host %@. HTTP STATUS: %ld", [url host], (long) statusCode];
                DDLogWarn(@"🔑 %@", msg);;
                error = [NSError errorWithDomain:@"TSAuthorization" code:-1 userInfo:@{NSLocalizedDescriptionKey:msg}];
                TSAuthorizationEvent *event = [[TSAuthorizationEvent alloc] initWithError:error];
                failure(event);
            }
        }
```

**Comment by christocracy** [comment]
```diff
switch(statusCode) {
+            case 200:
+            case 201: {
                .
                .
                .
```

Fix released to [`4.3.2`](https://pub.dev/packages/flutter_background_geolocation/versions/4.3.2)

**Comment by christocracy** [comment]
Confirmation of fix, please.
